databricks_cli/pipelines/cli.py

```diff
@@ -59,8 +69,19 @@ def deploy_cli(api_client, spec_arg, spec):
         raise RuntimeError('The spec should be provided either by an option or argument')
     src = spec_arg if bool(spec_arg) else spec
     spec_obj = _read_spec(src)
+    if 'id' not in spec_obj:
+        pipeline_id = str(uuid.uuid4())
+        click.echo("Updating spec at {} with id: {}".format(src, pipeline_id))
+        spec_obj['id'] = pipeline_id
+        _write_spec(src, spec_obj)
+    _validate_pipeline_id(spec_obj['id'])
```

Comment:
Why are we validating for deploy and get, but not reset and delete?

Reply:
all other methods except deploy call `_get_pipeline_id` which calls `_validate_pipeline_id`. So only deploy needs to call validate explicitly, all commands are covered